src/pages/Profile.js

```diff
@@ -3,38 +3,62 @@ import firebase from "firebase";
 import { StyleSheet, Text, View, Image, TouchableOpacity } from "react-native";
 import { Actions } from "react-native-router-flux";
 
+
 export default class Profile extends Component {
-  state = {
-    name: "",
-    info: "What do you do?",
-    description: "Here goes a brief description of you!",
-    edit: "Edit my profile"
-  };
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      profile: [],
+      name: "User",
+      info: "What do you do?",
+      description: "Here goes a brief description of you",
+      edit: "Edit my profile"
+    };
+  }
 
-  componentDidMount = () => {
-    
-      const { name } = firebase.auth();
-      this.setState({ name:name });
-    
-    
-  };
+  componentDidMount() {
+    const { currentUser } = firebase.auth();
+
+    firebase
+      .database()
+      .ref(`users/${currentUser.uid}/profile/`)
+      .on('value', snapshot => {     
+         var data = snapshot.val();
+        this.setState({
+          name: data.name,
+          info: data.info,
+          description: data.description
+        })
+      });
+  }
+
+  signOut() {
```

Comment:
using that format will give you problems later if you try to get the context with `this`. I recommend you to replace it with 
```suggestion
  signOut = () => {
```